Source/artifacts/ReadModelInquirer.js

```diff
@@ -0,0 +1,68 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Dolittle. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import { Folders } from '../Folders';
+import fs from 'fs';
+import global from '../global'
+
+const inquirer = require('inquirer');
+
+const _folders = new WeakMap();
+const _fileSystem = new WeakMap();
+
+export class ReadModelInquirer {
+
+    /**
+     * Constructor
+     * @param {Folders} folders
+     * @param {fs} fileSystem
+     */
+    constructor(folders, fileSystem) {
+        _folders.set(this, folders);
+        _fileSystem.set(this, fileSystem);
+    }
+    /**
+     * Gets the inquirer.js prompt answers based on the language
+     * @param {String} language
+     * @returns {Promise<any>} The answers
+     */
+    promptUser(language){
+        if (language === 'csharp') {
+            return this._getCSharpPrompt()
+        }
+    }
+
+    _getCSharpPrompt() {
+        const namespace = global.createCSharpNamespace(process.cwd(), global.getNearestCsprojFile());
```

Comment:
Using process directly will make automated specs virtually impossible to write consistently. I suggest introducing a context object that can can be taken dependencies on and set up during globals setup. In it we could have the current working directory as a propery.